mmf/models/base_model.py

```diff
@@ -213,15 +213,27 @@ def format_for_prediction(self, results, report):
 
     @classmethod
     def from_pretrained(cls, model_name_or_path, *args, **kwargs):
-        if not PathManager.isfile(model_name_or_path):
+        # Check if the path exists, if not it is pretrained, otherwise,
+        # we will try to load the checkpoint from the path
+        if not PathManager.exists(model_name_or_path):
             model_key = model_name_or_path.split(".")[0]
             model_cls = registry.get_model_class(model_key)
             assert (
                 model_cls == cls
             ), f"Incorrect pretrained model key {model_name_or_path} "
             "for class {cls.__name__}"
         output = load_pretrained_model(model_name_or_path, *args, **kwargs)
-        config, checkpoint = output["config"], output["checkpoint"]
+        config, checkpoint, full_config = (
+            output["config"],
+            output["checkpoint"],
+            output["full_config"],
+        )
+
+        # Save original config for state reset later
+        config_temp_holder = registry.get("config")
+        # Register full config from checkpoint when loading the model
+        registry.register("config", full_config)
```

Comment:
Thanks for fixing! Should we register another param in the registry `model_config` that is different so that not to override the same thing which might cause externalities?

Reply:
IMO, we should just keep one config in registry that is the global config to avoid any confusions. `model_config` can anyways be accessed from global config.